kapitan/remoteinventory/fetch.py

```diff
@@ -38,7 +46,15 @@ def fetch_inventories(inventory_path, target_objs, pool):
             for inv in inventories:
                 inv_type = inv["type"]
                 source_uri = inv["source"]
-                if source_uri in cached.inv_sources:
+                source_hash = hashlib.sha256(source_uri.encode())
+                # hashing the source and subdir together for git sources
+                # as different inventory items can have the same git uri
+                try:
+                    subdir = inv["subdir"]
+                    source_hash.update(subdir.encode())
+                except KeyError:
```

Comment:
before the `pass` make sure to always add a logger.debug("") message to make debugging easier. That is then displayed when devs run kapitan with `--verbose`